python/TestHarness/schedulers/RunPBS.py

```diff
@@ -33,21 +33,30 @@ def getBadKeyArgs(self):
     def hasTimedOutOrFailed(self, job_data):
         """ use qstat and return bool on job failures outside of the TestHarness's control """
 
-        KNOWN_EXITCODES = { '271' : 'JOB_EXEC_KILL_WALLTIME 271',
-                            '-24' : 'JOB_EXEC_KILL_NCPUS_BURST -24',
-                            '-25' : 'JOB_EXEC_KILL_NCPUS_SUM -25',
-                            '-26' : 'JOB_EXEC_KILL_VMEM -26',
-                            '-27' : 'JOB_EXEC_KILL_MEM -27',
-                            '-28' : 'JOB_EXEC_KILL_CPUT -28',
-                            '-29' : 'JOB_EXEC_KILL_WALLTIME -29' }
+        PBS_EXITCODES = { '271' : 'JOB_EXEC_KILL_WALLTIME 271',
+                          '-24' : 'JOB_EXEC_KILL_NCPUS_BURST -24',
+                          '-25' : 'JOB_EXEC_KILL_NCPUS_SUM -25',
+                          '-26' : 'JOB_EXEC_KILL_VMEM -26',
+                          '-27' : 'JOB_EXEC_KILL_MEM -27',
+                          '-28' : 'JOB_EXEC_KILL_CPUT -28',
+                          '-29' : 'JOB_EXEC_KILL_WALLTIME -29' }
 
-        launch_id = job_data.json_data.get(job_data.job_dir,
-                                           {}).get(job_data.plugin,
-                                                   {}).get('ID', "").split('.')[0]
+        PBS_STATUSES = { 'R' : 'RUNNING',
+                         'F' : 'FINISHED',
+                         'Q' : 'QUEUED',
+                         'E' : 'EXITING'}
```

Comment:
A beautiful corner case I saw during my testing. E... PBS was finishing up (exiting). I assume this means it was busy writing output.

Rather than happenstance, I should see what other possible status PBS can have.